OpenAssetIO/OpenAssetIO
# Add Python `.dist-info`

## Description

Closes #1088
- #1088

Add a `.dist-info` directory when `cmake --install`ing the project with Python enabled. This effectively prevents a `pip install` from overwriting it, with caveats (see below).


## Reviewer Notes

If this looks good, then I propose to simply duplicate it in OpenAssetIO-MediaCreation, rather than make it generic and portable and figure out a way for projects to share CMake libraries.  If we had more target projects, or there was expectation this feature would change frequently, then putting the effort in would be worthwhile.  But not for the sake of two projects.

The CI is structured differently than the rest of our workflows. For simplicity, it uses `bash` on all platforms, with the knock on effect of using the default CMake generator for the platform (see commit), and is roughly what I propose in #1114 .

~~CI only tests the effect we want, rather than deep tests of the `dist-info` structure. I propose to do that as part of #838, if needed.~~

## Test Instructions

~~As noted above, CI tests are limited to checking that `pip install` fails as expected. You may wish to manually inspect the contents of the installed files under `dist-info`.~~

`pip` will warn when installing `openassetio` into an environment where the `PYTHONPATH` contains another copy of `openassetio`, e.g.

> Not uninstalling openassetio at [...], outside environment

but will still install a competing copy.

However, if `openassetio` is installed into the Python environment directly, the we get an error, e.g.

> ERROR: Cannot uninstall openassetio 1.0.0a14, RECORD file not found. Hint: The package was installed by cmake.

and pip fails to overwrite the CMake installed `openassetio`.
